### PR TITLE
Launching a Site on Altis docs

### LINKED ISSUE #125: Provide Launch documentation
Public documentation for the process of launching a site on Altis, including the steps that need to take place.

This should be a guide in the documentation for prelaunch and launch day steps. Additionally, noting the ways you can get support and linking to https://docs.altis-dxp.com/nightly/guides/getting-help-with-altis/

The original description here has a good start, but may need updates or changes since it was first written.

----

*Original description follows:*

I think this is a mix of HM Engineering Handbook documentation and Altis documentation.

It would be interesting to provide a complete guide of how to launch projects by following a list that, based on HM experience, is something that anyone should follow and plan before a launch of a website or a project.

We have a bunch of examples but here are some interesting ones (sorry for the private links):

- https://github.com/humanmade/concentra/issues/234
- https://github.com/humanmade/hosting-support/issues/651

I'm going to brainstorm a bit here, sorry for that. Some steps/recommendations that I have in mind:

## Planning

- Create a planning issue in the project repository.
- Set a launch date
- Set a date to freeze the code (5 days is fine? Does it depend on the project size?)
- What part of the team is needed during the deploy?
- Is the website needed to be under maintenance?
- Set a date to freeze the content if needed.
- When is a good date to stop adding new features?
- Create the PR against `master`. Make sure that there are no conflicts and that there won't be errors after the merge.
- Get the PR approved.
- Create backups of all the environments.
- Do you need to move content from staging/development? Exporting the content in XML format and importing it by using a `wp import` command.
- Make sure that you can ssh to all the environments.

## Launch
- Are you moving environments? If so, would be better if a Cloud Engineer do it for you? Uploads folders can be massive so something could happen when pushing that button.

## Post-launch
- Do you need to set the default branch back to `master` in your repository?
- Check Search Console, Analytics or any similar service for errors after the deploy is done.
- Do you need to sync staging/development again?

## Servers related tasks

- Open a ZenDesk issue from Altis Dashboard to notify the server team with the full plan.
- Always notice the servers team, highlight the date if this is gonna be held during a weekend.
- Does the launch need any DNS or domain changes?
- Any redirections needed?
- Is sync between staging and production necessary?
- If it's a website that is already in production and the current project will need new content curation, will the client do it in production after the code is deployed? Will it do it in staging? Do you need to flip staging and production in that case? How is done in Altis/HM? **Update**. Don't do this, do not create content in staging.
- If you switch environments, make sure that the site is public after that and that the new staging/development environment is private (is this handled by Altis in any way?)
- Make sure that the deployed branch is going to be the correct one.
- If an SSL certificate is needed, does the server team have it?

With all this in mind I think we could create a good guide to follow when a launch is going to be done shortly. It will help project managers or developers to follow a path and make sure that they have everything they need.

# Launch checklist
This is copied and pasted from a launch list in https://github.com/humanmade/hosting-support
 
Expected Date/Time of Launch:

### Launch checklist for Project lead:

- Are we deploying the correct branch?
- Has AWS SES been properly setup?
- Content freeze?
- Client ready for DNS switches?
- Ensured site HTTPS readiness?

### Launch checklist for Servers team:

- Are we deploying the correct branch?
- Has AWS SES been properly setup?
- Have you selected the appropriate instance type?
- Has the database been synced?
- Have images been synced?
- Content freeze?
- Double checked the route53 entries are pointed to the correct CDN?
- SSL certificate loaded?

# New site checklist

Again, this is copied and pasted from https://github.com/humanmade/hosting-support

## PM/AM

- Readiness time-frame(s)
- Name of client:
- Main client business region:
- On-going Client contact lead:
- All domains to be hosted by our infrastructure. Enter as a list below:
  -
- What domains will email need to be sent from? Enter as a list below:
  -
- For each site in the network, what will be the FROM email? Enter as a list below:
  -
- Github repository:
- Github repository deploy branch name per environment:
- Existing database, if it exists:
- Existing uploads, if they exist:
- If this site is going to be running over HTTPS:
  - [ ] Client will be providing SSL certicate(s)
  - [ ] Human Made is responsible for generating the SSL Certificate
- Who controls the DNS for the client?
  - [ ] Human Made
  - [ ] Client
- Requirements for Elasticsearch:
- Projected traffic:
- Are there specific cookies that need to be whitelisted outside of the following patterns `comment_ hm_ wordpress_ wp- wp_`?
- Any other specifications or challenges?

## Cloud Engineers

- Nameserver records have been provided to the client (if we're managing their DNS).
- AWS SES domain verification DNS records have been provided to the client or entered into Route 53.
  - [ ]
- AWS SES Sending domains have been verified in SES.
  - [ ]
- AWS SES Sending domains have been configured in the application code.
- ACM Certificate DNS records have been provided to the client or entered into Route 53
  - [ ]
- ACM Certificate has been issued.
  - [ ]
- ACM Certificate has been attached to the environment.
  - [ ]
- [ ] Existing database, if any, has been imported into the environment.
- [ ] Existing uploads, if any, have been imported into the environment.
- [ ] Database has been search/replaced with the environment-specific domain as the replacement.

### PR DESCRIPTION
**Pull request by jazzsequence**
Adds documentation for launching a site.
fixes #125

### CODE REVIEW

**Comment by jazzsequence** [comment]
First pass at trying to combine the various bits and pieces of information from the original ticket as well as current launch support tickets.

**Comment by roborourke** [comment]
On my initial read-through I think this is very much a mixture of client facing and internal docs. Linking to terraform-app-stack for instance - that's a private repo and would provide no value for a client team.

I can see the value in having a big checklist of things in GitHub as you can actually tick them off but there's a question of scale and how we want to operate in future.

This approach is easy when clients have their repo under our organisation but that's not what we want as the norm, therefore we can expect only the `humanmade-cloud` GH account to have access for deployments and it'll be a pain for a cloud team member to be logging in and updating their bits and pieces.

We need to work out what's specific to the client and what's specific to the product team in terms of actions to take and only document client tasks here, noting when to hand off to the product team. In lieu of anything better for tracking I think rather than an issue in the project repo the client should manage this process through a support ticket in Zendesk.

File: other-docs/guides/launching-a-site-on-altis.md [hunk 1]
```diff
@@ -0,0 +1,181 @@
...
+- [ ] ACM Certificate DNS records have been provided to the client or entered into Route 53
+- [ ] ACM Certificate has been issued.
+- [ ] ACM Certificate has been attached to the environment.
+- [ ] Existing database, if any, has been imported into the environment.
+- [ ] Existing uploads, if any, have been imported into the environment.
+- [ ] Database has been search/replaced with the environment-specific domain as the replacement.
+
+<!-- For each environment that needs to be created or migrated, include a checklist like the following: -->
+
+### domain-dev
```
Comment (roborourke) [inline other-docs/guides/launching-a-site-on-altis.md:106]: These sections should just be internal docs replaced by a note to say that once the zendesk ticket has been created a cloud engineer will respond to that ticket with the following information:

specific info depending on some of the above, eg. are we managing their DNS

File: other-docs/guides/launching-a-site-on-altis.md [hunk 2]
```diff
@@ -0,0 +1,181 @@
...
+- [ ] **domain.com:** some-email@domain.com
+- [ ] etc.
+
+## Environment
+* **GitHub repository:** https://github.com/humanmade/{projectname}
+* **GitHub repository deploy branches:**
+  * **Production:** `main`
+  * **Staging:** `staging`
+  * **Development:** `dev`
+  * **Altis integration branch:** `altis` <!-- If applicable -->
```
Comment (roborourke) [inline other-docs/guides/launching-a-site-on-altis.md:80]: Seems to be more a migration specific thing so we could dump this

File: other-docs/guides/launching-a-site-on-altis.md [hunk 3]
```diff
@@ -0,0 +1,181 @@
...
+- [ ] etc.
+
+**For each site in the network, what will be the FROM email?
+<!-- Enter as a list -->
+- [ ] **Network Admin:** noreply@domain.com
+- [ ] **domain.com:** some-email@domain.com
+- [ ] etc.
+
+## Environment
+* **GitHub repository:** https://github.com/humanmade/{projectname}
```
Comment (roborourke) [inline other-docs/guides/launching-a-site-on-altis.md:75]:
```suggestion
* **GitHub repository:** https://github.com/{owner}/{repo}
```

File: other-docs/guides/launching-a-site-on-altis.md [hunk 4]
```diff
@@ -0,0 +1,181 @@
...
+* **GitHub repository deploy branches:**
+  * **Production:** `main`
+  * **Staging:** `staging`
+  * **Development:** `dev`
+  * **Altis integration branch:** `altis` <!-- If applicable -->
+* [ ] **Database exists in Altis Dashboard?**
+  * [ ] If not, provide a link to the database to import:
+* [ ] **Uploads exist in Altis Dashboard?**
+  * [ ] If not, provide a link to the uploads to import:
```
Comment (roborourke) [inline other-docs/guides/launching-a-site-on-altis.md:84]: Update to just request this information eg. the where, if it exists - need to check whether we actually do the initial import / transfer

File: other-docs/guides/launching-a-site-on-altis.md [hunk 5]
```diff
@@ -0,0 +1,181 @@
...
+  * **Staging:** `staging`
+  * **Development:** `dev`
+  * **Altis integration branch:** `altis` <!-- If applicable -->
+* [ ] **Database exists in Altis Dashboard?**
+  * [ ] If not, provide a link to the database to import:
+* [ ] **Uploads exist in Altis Dashboard?**
+  * [ ] If not, provide a link to the uploads to import:  
+* **Who is providing the SSL certificate(s) (Altis or client)?:** 
+* **Who controls the DNS for the client (Altis or client)?:**
+* **Special requirements or considerations for ElasticSearch:**
```
Comment (roborourke) [inline other-docs/guides/launching-a-site-on-altis.md:87]: Let's drop this! We don't want to invite any requests for setting snowflakey environments

File: other-docs/guides/launching-a-site-on-altis.md [hunk 6]
```diff
@@ -0,0 +1,181 @@
...
+* **Projected traffic:** 
+* **Specific cookies that need to be allow-listed outside these patterns `comment_`, `hm_`, `wordpress_`, `wp-`, `wp_`, `altis_`:**
+* **Any other specifications or challenges?**
```
Comment (roborourke) [inline other-docs/guides/launching-a-site-on-altis.md:90]: Make a reference the cloud env limitations doc

File: other-docs/guides/launching-a-site-on-altis.md [hunk 7]
```diff
@@ -0,0 +1,181 @@
+# Launching a Site on Altis
+
+The process for launching a site on Altis varies depending on the type of project. Nonetheless, there are common steps that need to be done no matter the context and this guide will walk you through what needs to be done in preparation for your site launch.
+
+## Planning
+
+Create a launch issue in the project repository. This issue should include:
```
Comment (roborourke) [inline other-docs/guides/launching-a-site-on-altis.md:7]: We should note advice on setting require login to true for the production environment until after the domains and everything have been setup and migrated to, then as a last step, require login should be removed, committed and deployed

File: other-docs/guides/launching-a-site-on-altis.md [hunk 8]
```diff
@@ -0,0 +1,181 @@
...
+- [ ] Update the CloudFormation stack template to retain CloudFront distribution.
+- [ ] Delete Cloudformation stack once after the client QA.
+
+```
+
+## Post-launch
+
+After the site launch is completed, there may be some items left to clean up or check.
+
+- Do you need to set the default branch back to `main` in your repository?
```
Comment (roborourke) [inline other-docs/guides/launching-a-site-on-altis.md:179]: Might be related to the migration branch note above? Not sure this is necessary

Comment (jazzsequence) [reply other-docs/guides/launching-a-site-on-altis.md:96]: That's probably accurate. New sites wouldn't likely need to switch deploy branches.

File: other-docs/guides/launching-a-site-on-altis.md (hunk 1 above)
Comment (jazzsequence) [reply other-docs/guides/launching-a-site-on-altis.md]: Added a new section between submitting a support ticket and launch that sort of covers this a bit ("the Altis Support team will work on setting up your environments and be in touch with your team"). Let me know if you think anything additional is needed here.

**Comment by jazzsequence** [comment]
Second pass at the doc with all the internal stuff moved to humanmade/product-dev#732.

I don't have anything around actual "launch" -- should we have a section about that? What would go in there?

File: other-docs/guides/launching-a-site-on-altis.md [hunk 9]
```diff
@@ -0,0 +1,126 @@
...
+
+The process for launching a site on Altis varies depending on the type of project. Nonetheless, there are common steps that need to be done no matter the context and this guide will walk you through what needs to be done in preparation for your site launch.
+
+## Planning
+
+Create a launch issue in the project repository. This issue should include:
+
+* Client name
+* Client business region
+* Client contact
```
Comment (roborourke) [inline other-docs/guides/launching-a-site-on-altis.md:11]: This could be followed by an internal team for a given client in which case this isn't needed. Additionally I think now that this issue isn't something the product team would need to interact with directly this information should be implicit from both the project team and Altis side so we can drop it.

File: other-docs/guides/launching-a-site-on-altis.md [hunk 10]
```diff
@@ -0,0 +1,126 @@
...
+* Launch date
+* Feature freeze date
+* Code freeze date
+* Content freeze date (if migrating an existing site)
+* Team members needed for the deploy
+* Domains hosted on Altis
+* Domain(s) used for email
+* FROM address for each site in the network
+* GitHub repository
```
Comment (roborourke) [inline other-docs/guides/launching-a-site-on-altis.md:20]: If the issue is going on the project's own repo this should be implicit too.

File: other-docs/guides/launching-a-site-on-altis.md [hunk 11]
```diff
@@ -0,0 +1,126 @@
...
+* Deploy branch name per environment
+* A launch checklist containing pre-launch, launch day and post-launch tasks (examples can be found below)
+* Any additional notes or complexities for the project launch
+
+An [example checklist](#Example-launch-checklist) for your GitHub issue can be found below. This issue is primarily for _internal project use_ and compiling all the necessary information. 
+
+Once the GitHub issue is created, this information can then be copied into [a support ticket in the Altis Dashboard](https://docs.altis-dxp.com/guides/getting-help-with-altis/) to alert the Altis Support Team.
```
Comment (roborourke) [inline other-docs/guides/launching-a-site-on-altis.md:27]: Maybe offer some advice on the type of ticket and a suitable title, thinking it'll be a "Task" and maybe "Site Launch Preparation" as the title.

File: other-docs/guides/launching-a-site-on-altis.md [hunk 12]
```diff
@@ -0,0 +1,126 @@
...
+After you have submitted your support request in the Altis Dashboard, the Altis Support team will work on setting up your environments and be in touch with your team. When your new cloud environments are set up, you will be contacted and can begin deploying your code to those new environments.
+
+Once you have environments set up, you should restrict access to them by [requiring login](https://docs.altis-dxp.com/security/require-login/), forcing [PHP basic authentication](https://docs.altis-dxp.com/security/php-basic-auth/), or both. In your Altis configuration file (`composer.json`), add the following to require login both and PHP authentication. You may omit PHP authentication, but we recommend having at least one of these enabled for all non-production and pre-launch sites. You can override the require login setting to not require login on local environments. PHP authentication is disabled by default on local environments. Refer to the documentation pages for more information about Require Login and PHP Basic Authentication.
+
+```json
+"altis": {
+	"modules": {
+		"security": {
+			"require-login": true,
+			"php-basic-auth": {
```
Comment (roborourke) [inline other-docs/guides/launching-a-site-on-altis.md:102]: Worth noting right now you can't use PHP Basic Auth on a production environment at all, at least not without messing around with the `hmauth_filter_dev_env` filter to add production as a dev environment. I'd just leave the example and advice as recommending requiring login for now.

File: other-docs/guides/launching-a-site-on-altis.md [hunk 13]
```diff
@@ -0,0 +1,126 @@
...
+
+```
+
+## Preparing Altis Cloud Environments
+After you have submitted your support request in the Altis Dashboard, the Altis Support team will work on setting up your environments and be in touch with your team. When your new cloud environments are set up, you will be contacted and can begin deploying your code to those new environments.
+
+Once you have environments set up, you should restrict access to them by [requiring login](https://docs.altis-dxp.com/security/require-login/), forcing [PHP basic authentication](https://docs.altis-dxp.com/security/php-basic-auth/), or both. In your Altis configuration file (`composer.json`), add the following to require login both and PHP authentication. You may omit PHP authentication, but we recommend having at least one of these enabled for all non-production and pre-launch sites. You can override the require login setting to not require login on local environments. PHP authentication is disabled by default on local environments. Refer to the documentation pages for more information about Require Login and PHP Basic Authentication.
+
+```json
+"altis": {
```
Comment (roborourke) [inline other-docs/guides/launching-a-site-on-altis.md:98]: I know it gets deeply nested but I'm trying to keep the example JSON configs complete so the full `{ "extra": { "altis" : { ... } } }`, at the very least this should be valid JSON so the code highlighting doesn't get confused.

File: other-docs/guides/launching-a-site-on-altis.md [hunk 14]
```diff
@@ -0,0 +1,126 @@
...
+# Goal
+
+<!-- A brief overview of the launch objective. Is it a migration? Is it a new site? -->
+
+## Pre-launch checklist
+<!-- To be filled out by the project team/lead. Not needed for Altis Support ticket. -->
+- [ ] Are we deploying the correct branch?
+- [ ] Has AWS SES been setup?
+- [ ] Is a content freeze active?
+- [ ] Is the client ready for DNS switches?
```
Comment (roborourke) [inline other-docs/guides/launching-a-site-on-altis.md:41]: Mentioning the client again but this might be the client doing the launch themselves, "is the site ready for the DNS switch over?"

File: other-docs/guides/launching-a-site-on-altis.md [hunk 15]
```diff
@@ -0,0 +1,126 @@
...
+- [ ] Site is ready for HTTPS?
```
Comment (roborourke) [inline other-docs/guides/launching-a-site-on-altis.md:42]: We're removing this right?

File: other-docs/guides/launching-a-site-on-altis.md [hunk 16]
```diff
@@ -0,0 +1,126 @@
...
+- [ ] Are any redirections needed?
+
+## Domains hosted on Altis
+<!-- List of domains that will be on the stack -->
```
Comment (roborourke) [inline other-docs/guides/launching-a-site-on-altis.md:46]: I think this just needs to be custom domains, so `<project>-staging.altis.cloud` and `<project>-dev.altis.cloud` are the default domains we'll provide for those environments.

Lets keep the focus on the production environment for this guide and issue template so remove the staging and development sections below.

File: other-docs/guides/launching-a-site-on-altis.md [hunk 17]
```diff
@@ -0,0 +1,126 @@
...
+- **Network Admin:** noreply@domain.com
+- **domain.com:** some-email@domain.com
+- etc.
+
+## Environment
+* **GitHub repository:** https://github.com/{owner}/{repo}
+* **GitHub repository deploy branches:**
+  * **Production:** `main`
+  * **Staging:** `staging`
+  * **Development:** `dev`
```
Comment (roborourke) [inline other-docs/guides/launching-a-site-on-altis.md:80]: Shouldn't need this for the cloud team if we're assuming the production environment will exist already.

File: other-docs/guides/launching-a-site-on-altis.md (hunk 10 above)
Comment (jazzsequence) [reply other-docs/guides/launching-a-site-on-altis.md:20]: That might not be obvious if we're telling them to make the GitHub issue to gather information and then submit a Zendesk ticket from the Altis Dashboard. If we're not seeing this issue in GitHub, do you think we should still pull out the repo link?

File: other-docs/guides/launching-a-site-on-altis.md (hunk 14 above)
Comment (jazzsequence) [reply other-docs/guides/launching-a-site-on-altis.md:41]:
```suggestion
- [ ] Is the site ready for the DNS switch?
```

File: other-docs/guides/launching-a-site-on-altis.md (hunk 12 above)
Comment (jazzsequence) [reply other-docs/guides/launching-a-site-on-altis.md]: I forgot you had to filter to run it on production 👍
It's been a while since I looked at that code...

File: other-docs/guides/launching-a-site-on-altis.md (hunk 17 above)
Comment (jazzsequence) [reply other-docs/guides/launching-a-site-on-altis.md:75]: My understanding is that the support ticket that this would be a part of would be to initially set up the environments. Is that not what you were thinking?

File: other-docs/guides/launching-a-site-on-altis.md (hunk 11 above)
Comment (jazzsequence) [reply other-docs/guides/launching-a-site-on-altis.md:23]: Would they be able to use the Altis Dashboard to create a new site if one did not already exist or is that why we're assuming that the environments are already set up?

File: other-docs/guides/launching-a-site-on-altis.md (hunk 17 above)
Comment (jazzsequence) [reply other-docs/guides/launching-a-site-on-altis.md:75]: Yep, need to assume we've already created the environments at this point and we're just prepping for launch. 👍

File: other-docs/guides/launching-a-site-on-altis.md [hunk 18]
```diff
@@ -0,0 +1,104 @@
+# Launching a Site on Altis
+
+The process for launching a site on Altis varies depending on the type of project. Nonetheless, there are common steps that need to be done no matter the context and this guide will walk you through what needs to be done in preparation for your site launch.
+
+## Planning
+
+Create a launch issue in the project repository. This issue should include:
```
Comment (rmccue) [inline other-docs/guides/launching-a-site-on-altis.md:7]: IMO, this should be a "we recommend", because we don't necessarily require people to open issues.

File: other-docs/guides/launching-a-site-on-altis.md [hunk 19]
```diff
@@ -0,0 +1,104 @@
...
+
+* Launch date
+* Feature freeze date
+* Code freeze date
+* Content freeze date (if migrating an existing site)
+* Team members needed for the deploy
+* Domains hosted on Altis
+* Domain(s) used for email
+* FROM address for each site in the network
```
Comment (rmccue) [inline other-docs/guides/launching-a-site-on-altis.md:16]: Are these not the same thing?

File: other-docs/guides/launching-a-site-on-altis.md [hunk 20]
```diff
@@ -0,0 +1,104 @@
...
+
+
+```
+
+## Preparing Altis Cloud Environments
+After you have submitted your support request in the Altis Dashboard, the Altis Support team will work on setting up your environments and be in touch with your team. When your new cloud environments are set up, you will be contacted and can begin deploying your code to those new environments.
+
+Once you have environments set up, you should restrict access to them by [requiring login](https://docs.altis-dxp.com/security/require-login/), forcing [PHP basic authentication](https://docs.altis-dxp.com/security/php-basic-auth/), or both. In your Altis configuration file (`composer.json`), add the following to require login both and PHP authentication. You may omit PHP authentication, but we recommend having at least one of these enabled for all non-production and pre-launch sites. You can override the require login setting to not require login on local environments. PHP authentication is disabled by default on local environments. Refer to the documentation pages for more information about Require Login and PHP Basic Authentication.
```
Comment (rmccue) [inline other-docs/guides/launching-a-site-on-altis.md:73]: I would rephrase this to break it up a bit. Also, it's not "PHP authentication", it's "HTTP Basic authentication" (or just "Basic auth"); that might be something we need to correct in the other docs too?

```suggestion
Once you have environments set up, we recommend limiting access to your sites. By default, login will be required on your development and staging environments. Before your site launches, we also recommend requiring login for your production environment as well.

Logins can be enforced by [requiring user logins](https://docs.altis-dxp.com/security/require-login/), forcing [HTTP basic authentication](https://docs.altis-dxp.com/security/php-basic-auth/), or both.

In your Altis configuration file (`composer.json`), add the following to both require login and HTTP Basic authentication. You may omit Basic authentication, but we recommend having at least one of these enabled for all non-production and pre-launch sites. You can override the require login setting to not require login on local environments. Basic authentication is disabled by default on local environments. Refer to the documentation pages for more information about Require Login and Basic Authentication.
```

It seems to me like we should also just enable require-login on prod, so that when the site launches, it's easy to just remove those lines?

File: other-docs/guides/launching-a-site-on-altis.md [hunk 21]
```diff
@@ -0,0 +1,104 @@
...
+						}
+					}
+				}
+			}
+		}
+	}
+}
+```
+
+## Post-launch
```
Comment (rmccue) [inline other-docs/guides/launching-a-site-on-altis.md:98]: These steps are really just before the launch happens.

```suggestion

## Preparation for launch
```

**Review by rmccue — CHANGES_REQUESTED** [review]
Looks good, although I would also expect to see info about how the actual launch process works; i.e. content migration, DNS switchover, etc

File: other-docs/guides/launching-a-site-on-altis.md (hunk 19 above)
Comment (jazzsequence) [reply other-docs/guides/launching-a-site-on-altis.md:16]: Not necessarily. There are cases that I am currently aware of where there are multiple domains, but the FROM address for all of them is a single email address. For example if the domains were:

``` 
 - happyfunland.com
 - happyfunlandnews.com
 - happyfunlandgames.com
 - cruisehappyfun.com
 - etc
```

and they wanted the From email from all of those to be `noreply@happyfunland.com`.

File: other-docs/guides/launching-a-site-on-altis.md (hunk 20 above)
Comment (jazzsequence) [reply other-docs/guides/launching-a-site-on-altis.md]: > It seems to me like we should also just enable require-login on prod, so that when the site launches, it's easy to just remove those lines?

Could do. I can spin up an issue for it.

File: other-docs/guides/launching-a-site-on-altis.md [hunk 22]
```diff
@@ -0,0 +1,128 @@
...
+wp search-replace domain-production.altis.cloud domain.com --all-tables --network --url=domain-production.altis.cloud
+```
+
+This step will need to be repeated for any subdomains you have. Be sure to flush the cache when you are done, otherwise the old URLs will still be saved in the object cache.
+
+```bash
+wp cache flush
+```
+
+After the database is updated and the site is ready to go, contact Altis Support to let them know that you are ready for the DNS switch to your new site.
```
Comment (jazzsequence) [inline other-docs/guides/launching-a-site-on-altis.md:126]: Anything else here? Do we need to specify contacting Altis Support via their launch ticket/Altis Dashboard vs Slack?

**Comment by jazzsequence** [comment]
Anything else this should cover?

**Comment by rmccue** [comment]
Copying across a couple items from humanmade/altis-cloud#222:
* Required lead time on performing migrations
* Scheduling expectations and how we deal with exceptions

**Comment by jazzsequence** [comment]
@rmccue Pushed up some changes that add some "setting expectations" notes around migrations. I'm mostly just making stuff up, so if there are specific policies that we want to communicate, I'm happy to change/update or review any internal docs we might have already.

File: other-docs/guides/launching-a-site-on-altis.md (hunk 22 above)
Comment (roborourke) [reply other-docs/guides/launching-a-site-on-altis.md:132]: I think using the launch ticket makes sense to track the whole process and act as a log we can show. It makes sense to add that as a suggestion.

Comment (jazzsequence) [reply other-docs/guides/launching-a-site-on-altis.md:132]: @roborourke Is that not sufficiently covered [here](https://github.com/humanmade/altis-documentation/pull/222/files#diff-6cf7fa890a1e26dae62794924a67bdaf3f4bd7fee7a8f72479a6f4a4539e5ac6R23)?

**Comment by jazzsequence** [comment]
Added new intro paragraph for the "Site Launch" section and updated headings so they are capitalized consistently

File: other-docs/guides/launching-a-site-on-altis.md (hunk 22 above)
Comment (roborourke) [reply other-docs/guides/launching-a-site-on-altis.md:134]: Yeah true, staring at it too long!

**Review by roborourke — APPROVED** [review]
Don't just ship it, spaceship it 🚀
